app/src/androidTest/java/ch/epfl/sdp/blindly/fake_module/FakeUserRepositoryModule.kt

```diff
@@ -19,12 +38,158 @@ import javax.inject.Singleton
+            .setShowMe(showMe)
+            .setPassions(passions)
+            .setRadius(radius)
+            .setMatches(matches)
+            .setLikes(likes)
+            .setRecordingPath(recordingPath)
+            .setAgeRange(ageRange)
+            .build()
+        val fakeUserUpdated = User.Builder()
+            .setUid(uid)
+            .setUsername(usernameUpdated)
+            .setLocation(locationUpdated)
+            .setBirthday(birthday)
+            .setGender(genderUpdated)
+            .setSexualOrientations(sexualOrientationsUpdated)
+            .setShowMe(showMeUpdated)
+            .setPassions(passionsUpdated)
+            .setRadius(radiusUpdated)
+            .setMatches(matches)
+            .setLikes(likes)
+            .setRecordingPath(recordingPath)
+            .setAgeRange(ageRangeUpdated)
+            .build()
+    }
+
+    @Singleton
+    @Provides
+    open fun provideUserCache(): UserCache {
+        val userCache = mock(UserCache::class.java)
+        Mockito.`when`(userCache.get(TEST_UID)).thenReturn(fakeUser)
+        return userCache
+    }
+    @Singleton
+    @Provides
+    open fun provideFirebaseFirestore(): FirebaseFirestore {
+        val db = mock(FirebaseFirestore::class.java)
+
+        val taskCompletionSource = TaskCompletionSource<DocumentSnapshot>()
+        Handler(Looper.getMainLooper()).postDelayed({ taskCompletionSource.setResult(null) }, 1000L)
+        val successfulTask = taskCompletionSource.task
+        val collection = mock(CollectionReference::class.java)
+        val document = mock(DocumentReference::class.java)
+        Mockito.`when`(db.collection(USER_COLLECTION)).thenReturn(collection)
+        Mockito.`when`(
+            db.collection(USER_COLLECTION)
+                .document(TEST_UID)
+        ).thenReturn(document)
+        //TODO db.collection().document().get().await().toUser is a suspend function which is not the case of the real module
+        // Need to find a proper way to retrun the fakeUser after await()
+        Mockito.`when`(
+            db.collection(USER_COLLECTION)
+                .document(TEST_UID)
+                .get()
+        )
+            .thenReturn(successfulTask).then { fakeUser }
+
+        return db
+    }
+
     @Singleton
     @Provides
-    open suspend fun provideUserRepository(): UserRepository {
-        val userRepository = mock(UserRepository::class.java)
-        Mockito.`when`(userRepository.getUser(TEST_UID)).thenReturn(fakeUser)
-        Mockito.`when`(userRepository.refreshUser(TEST_UID)).thenReturn(fakeUser)
-        return userRepository
+    open fun provideUserRepository(): UserRepository {
+        return (Mockito.spy(object: UserRepository {
+            val db = HashMap<String, User>()
+            override suspend fun getUser(uid: String): User? {
+                return db.getOrDefault(uid, fakeUser)
+            }
+
+            override suspend fun removeMatchFromAUser(

```

Comment:
Woooow so this would enable us to test any function that calls the removeMatchFromAUser ? :D